jormungandr/src/network/mod.rs

```diff
@@ -161,7 +167,9 @@ impl GlobalState {
 
         let mut rng_seed = [0; 32];
         rand::thread_rng().fill(&mut rng_seed);
-        let prng = ChaChaRng::from_seed(rng_seed);
+        let mut prng = ChaChaRng::from_seed(rng_seed);
+
+        let keypair = NodeKeyPair::generate(&mut prng);
```

Comment:
Why do we use a seed?

Reply:
This is probably overkill (`ThreadRng` is a `CryptoRng` anyway), but the initialization of a secure RNG from seed was there for Poldercast. This key generation call just reuses the same RNG.

Reply:
I wrote it too fast: `CryptoRng` is not the trait that ensures unpredictability, but `ThreadRng` delivers that as well.

It may be useful, however, to have a single seeding point that can be overridden from the environment for test reproducibility purposes.

Reply:
I'm not sure I get it, but having a fixated seed pings something in my mind. I understand that for testing would be interesting. But maybe it would be safer to use `ChaChaRng::from_entropy`?

Reply:
I believe this is what the `ThreadRng` does under the hood when initializing. I can change this to streamline the code, `from_entropy` should be searchable enough to change to optionally deterministic seeding later.